examples/size_limiter.cpp

```diff
@@ -0,0 +1,29 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#include <iostream>
+
+#include "umpire/ResourceManager.hpp"
+#include "umpire/strategy/SizeLimiter.hpp"
+
+int main(int, char**)
+{
+  auto& rm = umpire::ResourceManager::getInstance();
+  auto alloc = rm.makeAllocator<umpire::strategy::SizeLimiter>(
+      "size_limited_alloc", rm.getAllocator("HOST"), 64);
+
+  void* data = alloc.allocate(1024);
+
+  return 0;
+}
```

Comment:
It would be SWEET if the example could also show how to:
`auto alloc = rm.makeAllocator<umpire::strategy::DynamicPool>(
      "host_simpool", rm.getAllocator("size_limited_alloc"));`